Refactor geographic gatherering

pnuu opened the pull request:
This PR refactors the code `gatherer.py` uses, deprecates `gatherer.py` in favor of more descriptive `geographic_gatherer.py` and does general clean-up of the code.

pnuu commented:
I think this has gotten far enough so that the first round of reviews are called for. I'll be continuing to comb through the code if I can spot something.

gerritholl commented:
Could you update the documentation too?

pnuu commented:
Will do! Forgot we had any, but you did add some a while back :grin:

gerritholl commented:
Will I be able to use the new `geographic_gatherer.py` as a drop-in replacement of `gatherer.py`, or will users need to make any other changes to their configuration?

pnuu commented:
It's a drop-in replacement. Basically I just renamed the script for clarity, and added a call from the old name to the new one. The major part is the internal restructuring, which shouldn't affect the usage.

mraspaud commented on bin/geographic_gatherer.py:
```diff
@@ -0,0 +1,111 @@
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Gather granule messages to send them in a bunch."""
```
This should be update to mention 'geographic'

pnuu replied:
Clarified the docstring.

mraspaud reviewed (approved):
LGTM

mraspaud commented on pytroll_collectors/tests/test_geographic_gatherer.py:
```diff
@@ -81,21 +81,23 @@ def setUp(self):
+
+        self.watchdog_trigger_patcher = patch('pytroll_collectors.geographic_gatherer.WatchDogTrigger')
+        self.WatchDogTrigger = self.watchdog_trigger_patcher.start()
         self.addCleanup(self.watchdog_trigger_patcher.stop)
```
```suggestion
        self.watchdog_trigger_patcher = patch_cleanup('pytroll_collectors.geographic_gatherer.WatchDogTrigger')
        
        def patch_cleanup(self, item):
            patcher = patch(item)
            patched = patcher.start()
            self.addCleanup(patcher.stop)
            return patched
        
```

mraspaud reviewed (approved):
LGTM
